nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -873,6 +873,13 @@ private Description checkReturnExpression(
         retExpr, methodSymbol, this, state);
     if (getMethodReturnNullness(methodSymbol, state, Nullness.NULLABLE).equals(Nullness.NULLABLE)) {
       return Description.NO_MATCH;
+    } else if (config.isJSpecifyMode()
+        && GenericsChecks.getGenericMethodReturnTypeNullness(
+                methodSymbol, ASTHelpers.getType(tree), state, config)
+            .equals(Nullness.NULLABLE)) {
+      // Get the Nullness if the Annotation is indirectly applied through a generic type if we
+      // are in JSpecify mode
+      return Description.NO_MATCH;
```

Comment:
This code won't work for the case when the lambda is written `x -> { return null; }`.  I updated the test and we get a false positive.  The issue is that with the `return null` case, the `tree` passed in is the `ReturnTree`, not the tree for the lambda itself.  One way to fix this would be to add another parameter `@Nullable Tree lambdaTree` to `checkReturnExpression` and then only run this case when `lambdaTree` is non-null.  The extant `tree` parameter can be renamed to `errorTree`, since its main purpose is for finding the right locations to report errors.

Reply:
I have a question regarding this, how would this work in the case of testNegative2? because we would still get a false positive there because we cannot extract the type in the lambda tree parameter is null right?